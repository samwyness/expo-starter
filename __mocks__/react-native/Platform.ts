export const mockPlatform = (os: string, version?: string | number) => {
  jest.doMock('react-native/Libraries/Utilities/Platform', () => ({
    ...jest.requireActual('react-native/Libraries/Utilities/Platform'),
    OS: os,
    select: (osList: Record<string, unknown>) => osList[os],
    Version: version,
  }));
};
