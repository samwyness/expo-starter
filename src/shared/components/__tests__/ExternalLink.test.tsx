import { render, userEvent } from '@testing-library/react-native';
import { renderRouter, screen } from 'expo-router/testing-library';
import { openBrowserAsync } from 'expo-web-browser';

import { ExternalLink } from '../ExternalLink';

import { mockPlatform } from '@/__mocks__/react-native/Platform';

jest.mock('expo-web-browser');

const mockOpenBrowserAsync = jest.mocked(openBrowserAsync);

describe(`ExternalLink`, () => {
  beforeEach(() => {
    mockOpenBrowserAsync.mockClear();
  });

  it(`renders correctly`, () => {
    // Arrange
    const tree = render(
      <ExternalLink href="some-href">Snapshot test!</ExternalLink>,
    ).toJSON();

    // Act

    // Assert
    expect(tree).toMatchSnapshot();
  });

  it(`should open the native browser with the correct href`, async () => {
    // Arrange
    renderRouter(['index']);
    render(<ExternalLink href="some-href">Snapshot test!</ExternalLink>);

    // Act
    await userEvent.press(screen.getByRole('link', { name: 'Snapshot test!' }));

    // Assert
    expect(mockOpenBrowserAsync).toHaveBeenCalledWith('some-href');
  });

  it(`should handle onPress correctly for web`, async () => {
    // Arrange
    mockPlatform('web', '1.0');

    renderRouter(['index']);
    render(<ExternalLink href="some-href">Snapshot test!</ExternalLink>);

    // Act
    await userEvent.press(screen.getByRole('link', { name: 'Snapshot test!' }));

    // Assert
    expect(mockOpenBrowserAsync).not.toHaveBeenCalled();
  });
});
