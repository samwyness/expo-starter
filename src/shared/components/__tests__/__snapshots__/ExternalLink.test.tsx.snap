// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink renders correctly 1`] = `
<Text
  href="/"
  onPress={[Function]}
  role="link"
>
  Snapshot test!
</Text>
`;
